pandas/tests/indexing/multiindex/test_setitem.py

```diff
@@ -10,133 +8,114 @@
-                )
-                check(
-                    target=df,
-                    indexers=((t, n), "X"),
-                    value=np.array(3),
-                    compare_fn=assert_equal,
-                    expected=3,
-                )
-
-                # GH5206
-                df = DataFrame(
-                    np.arange(25).reshape(5, 5),
-                    columns="A,B,C,D,E".split(","),
-                    dtype=float,
-                )
-                df["F"] = 99
-                row_selection = df["A"] % 2 == 0
-                col_selection = ["B", "C"]
-                with catch_warnings(record=True):
-                    df.ix[row_selection, col_selection] = df["F"]
-                output = DataFrame(99.0, index=[0, 2, 4], columns=["B", "C"])
-                with catch_warnings(record=True):
-                    tm.assert_frame_equal(df.ix[row_selection, col_selection], output)
-                check(
-                    target=df,
-                    indexers=(row_selection, col_selection),
-                    value=df["F"],
-                    compare_fn=tm.assert_frame_equal,
-                    expected=output,
-                )
-
-                # GH11372
-                idx = MultiIndex.from_product(
-                    [["A", "B", "C"], date_range("2015-01-01", "2015-04-01", freq="MS")]
-                )
-                cols = MultiIndex.from_product(
-                    [["foo", "bar"], date_range("2016-01-01", "2016-02-01", freq="MS")]
-                )
-
-                df = DataFrame(np.random.random((12, 4)), index=idx, columns=cols)
-
-                subidx = MultiIndex.from_tuples(
-                    [("A", Timestamp("2015-01-01")), ("A", Timestamp("2015-02-01"))]
-                )
-                subcols = MultiIndex.from_tuples(
-                    [("foo", Timestamp("2016-01-01")), ("foo", Timestamp("2016-02-01"))]
-                )
-
-                vals = DataFrame(
-                    np.random.random((2, 2)), index=subidx, columns=subcols
-                )
-                check(
-                    target=df,
-                    indexers=(subidx, subcols),
-                    value=vals,
-                    compare_fn=tm.assert_frame_equal,
-                )
-                # set all columns
-                vals = DataFrame(np.random.random((2, 4)), index=subidx, columns=cols)
-                check(
-                    target=df,
-                    indexers=(subidx, slice(None, None, None)),
-                    value=vals,
-                    compare_fn=tm.assert_frame_equal,
-                )
-                # identity
-                copy = df.copy()
-                check(
-                    target=df,
-                    indexers=(df.index, df.columns),
-                    value=df,
-                    compare_fn=tm.assert_frame_equal,
-                    expected=copy,
-                )
+        for index_fn in ("loc",):
```

Comment:
Can we just get rid of the loop altogether now?

Reply:
yes, there will probably be lots of clean-up opportunities which can be done as a follow-up.

the aim is to just remove for now, deprecations are the prioirty for 1.0 (https://github.com/pandas-dev/pandas/issues/27492#issue-470694636), and keeps the diff easier to review..

here, specificically, more pytest idiom would be nice, so a refactor to just to remove the loop could be just going down another rabbit hole. this pattern of a single loop is not unique to this test.

Reply:
@simonjayhawkins this pattern is fine